.github/workflows/curldemo.yaml

```diff
@@ -0,0 +1,45 @@
+name: curldemo
+
+on:
+  push:
+    tags:
+      - "v*"
```

Comment:
It's not connected to main release, so it is better to execute as reaction to file patch change, see https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#example-including-paths